### DISCUSSION TITLE
how to get octofarm installed on windows

### DISCUSSION THREAD

**Discussion by sidneyns** [OctoFarm/OctoFarm#753] [General]
Hello guys. 

I am trying to install octofarm on windows. I was able to get one computer a while ago to work, but trying to get it on the second is getting dificult. 

I am following chris basements on youtube walkthroug but on the octofarm installation is the part I am getting an issue. 

https://www.youtube.com/watch?v=9U-QTOmx49c&t=1254s

Apparently on the video, when you download octofarm on github, it supposed to also extract a folder called config, etc..  but when I search on githut it does not have this folder when I unzip the octofarm folder.  I was able to install mongo db and node js normally

not really sure where to get that version on the video or how to solve it 

Can anyone help?

Sidney

**Comment by NotExpectedYet** [comment]
> Hello guys.
> 
> I am trying to install octofarm on windows. I was able to get one computer a while ago to work, but trying to get it on the second is getting dificult.
> 
> I am following chris basements on youtube walkthroug but on the octofarm installation is the part I am getting an issue.
> 
> https://www.youtube.com/watch?v=9U-QTOmx49c&t=1254s
> 
> Apparently on the video, when you download octofarm on github, it supposed to also extract a folder called config, etc.. but when I search on githut it does not have this folder when I unzip the octofarm folder. I was able to install mongo db and node js normally
> 
> not really sure where to get that version on the video or how to solve it
> 
> Can anyone help?
> 
> Sidney

I'll get back to you with a proper reply as soon as I can but the config folder is no more. When you start up OF you will get a .env file now. I will get the old windows documentation ported over to docs.octofarm.net at the weekend for you :)

**Comment by sidneyns** [comment]
Thank you my friend

Em qui, 30 de set de 2021 02:31, James Mackay ***@***.***>
escreveu:

**Reply by NotExpectedYet** [reply]
> Thank you my friend Em qui, 30 de set de 2021 02:31, James Mackay ***@***.***> escreveu:

Welcome, 

If you've installed MongoDB, NodeJS, Git and pm2, then inside the OF folder you should just be able to run `npm start`

**Reply by NotExpectedYet** [reply]
Don't know if chris added pm2 but: 

`npm install -g pm2`
then: 
`npm start`

Should get that running :)

**Comment by sidneyns** [comment]
Thanks!

Did not install gt and pm2


Em qui, 30 de set de 2021 08:03, James Mackay ***@***.***>
escreveu:

**Comment by sidneyns** [comment]
Hot pm2 installed.
 I think it working... Many thanks

Em qui, 30 de set de 2021 09:15, Sidney Sato ***@***.***> escreveu:

**Comment by sidneyns** [comment]
*ca*n I use this tread to ask ... is there a easy way to update octofarm
without risk?

maybe by commands?

I am afraid to mess something up

**Reply by NotExpectedYet** [reply]
https://docs.octofarm.net/getting-started/octofarm-updating.html 

There's all information here :)

**Comment by sidneyns** [comment]
I think I have older version, there is no upgrade button

[image: image.png]

**Comment by sidneyns** [comment]
I have to do that CLI updating?

**Reply by NotExpectedYet** [reply]
If your on 1.1.13-hotfix then there's an update button but only when it's available. If you've installed the latest master you will be on 1.1.13-hotfix and there's no updates after that currently. When one is available you'll get a notification within the UI, or you can use the Check button on the system page to force it.

**Comment by sidneyns** [comment]
Hello James.

Not abusing your kindness.
The instance I installed with pm2, octofarm works, but when I try to access
my printers ( I think the database ) , it is really slow.

any tips?

thank you

Sidney

**Reply by NotExpectedYet** [reply]
It's a known issue on windows currently. It's the system info thing grabbing CPU/ Memory. 

There's currently a version on development I'm trying to get ready for release that fixes things. 

Your welcome to switch out to the development branch, but there may be bugs. I'm hoping "and praying" I'll have it ready by the weekend if you can live with it until then. 

Sorry about that, completely forgot to mention.

**Comment by sidneyns** [comment]
That is no problem.. your Tool is really good

Em qui, 21 de out de 2021 10:59, James Mackay ***@***.***>
escreveu:

**Comment by sidneyns** [comment]
Hello James

Did you guys fixed the windows issue on octofarm?

Thank you in advance

Sidney

**Reply by NotExpectedYet** [reply]
Hello Sidney, 

It's fixed in V1.2. Smashed in Canary :P. 

I'm just making sure everything is ready for V1.2 release. I'm looking about a week or maybe two as I'm back at work. Your welcome to try out one of the RC's if you get chance, I'd love some more confirmation feedback.

**Comment by sidneyns** [comment]
Will try.. How do I access RC?

Em seg., 10 de jan. de 2022 10:11, James Mackay ***@***.***>
escreveu:

**Reply by NotExpectedYet** [reply]
> Will try.. How do I access RC? Em seg., 10 de jan. de 2022 10:11, James Mackay ***@***.***> escreveu:
> […](#)
> Hello Sidney, It's fixed in V1.2. Smashed in Canary :P. I'm just making sure everything is ready for V1.2 release. I'm looking about a week or maybe two as I'm back at work. Your welcome to try out one of the RC's if you get chance, I'd love some more confirmation feedback. — Reply to this email directly, view it on GitHub <[#753 (reply in thread)](https://github.com/OctoFarm/OctoFarm/discussions/753#discussioncomment-1940114)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AR4X6Z5ZAEXESIZ2D4BH66LUVLLIFANCNFSM5FBAYRZA> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. You are receiving this because you authored the thread.Message ID: ***@***.***>

Have you installed with git? via git clone? 

If so

Open CMD in your octofarm directory and do the following. 

`pm2 delete OctoFarm`
`git fetch`
`git checkout development`
`git pull`
`npm install`

**Reply by NotExpectedYet** [reply]
I haven't checked Filament Manager, nor filament manager plugin support. Also the Bulk upload in file manager and the new multi-print option is currently broken just to warn. 

There's some UI bugs still as well, mostly just rounding on graphs. 

As far as I'm aware, monitoring and managing printers should be A.OK!
